aimmo-game/tests/functional/test_movements_in_map.py

```diff
@@ -0,0 +1,196 @@
+from unittest import TestCase
+
+from mock_world import MockWorld
+from simulation.location import Location
+from simulation import map_generator
+from tests.test_simulation.dummy_avatar import MoveEastDummy, MoveWestDummy, MoveNorthDummy, MoveSouthDummy, WaitDummy
+
+
+class TestMovementsInMap(TestCase):
+
+    SETTINGS = {
+        'START_HEIGHT': 50,
+        'START_WIDTH': 50,
+        'OBSTACLE_RATIO': 0,
+    }
+
+    def set_up_environment(self, dummy_list=None, location=Location(0, 0), map_generator_class=map_generator.Main):
+        """
+        Utility method for testing.
+        """
+        self.game = MockWorld(TestMovementsInMap.SETTINGS, dummy_list, map_generator_class)
+        self.game.game_state.add_avatar(1, None, location)
+        self.avatar = self.game.avatar_manager.get_avatar(1)
+
+    def test_movement_five_times_in_all_directions(self):
+        """
+        Moves the avatar to the edge of the map. Each time it moves the avatar 5 times from origin in all cardinal
+        directions.
+        """
+
+        # East.
+        self.set_up_environment([MoveEastDummy])
+        self.assertEqual(self.avatar.location, Location(0, 0))
+
+        for i in range(5):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertEqual(self.avatar.location, Location(5, 0))
+
+        # West.
+        self.set_up_environment([MoveWestDummy])
+        self.assertEqual(self.avatar.location, Location(0, 0))
+
+        for i in range(5):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertEqual(self.avatar.location, Location(-5, 0))
+
```

Comment:
it's just that it looks like it could easily be templated, instead of writing the same code over and over again?